### PR TITLE
feat(Reveal): Adds Reveal component

### PR DESCRIPTION
**Pull request by HHogg**
_The animations are actually smoother than in the GIF_
reveal

### CODE REVIEW

File: src/components/reveal/Reveal.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+export default class Reveal extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    space: PropTypes.string,
+    visible: PropTypes.bool.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state ={
```
Comment (thatguynamedandy) [inline src/components/reveal/Reveal.js]: `={` accidental emoji.

File: src/components/reveal/example/reveal.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    super(props);
+    this.state = {
+      visible: false,
+    };
+  }
+
+  handleOnClick() {
+    const { visible } = this.state;
+
+    this.setState({ visible: !visible });
```
Comment (thatguynamedandy) [inline src/components/reveal/example/reveal.js]: I'm starting to favour the functional approach for this kind of change, what do you think?
```
this.setState(({ visible }) => ({ visible: !visible }));
```

Comment (HHogg) [inline src/components/reveal/example/reveal.js]: Yeah I prefer that too :+1: Ok to follow up with a whole library change separately?

Comment (thatguynamedandy) [inline src/components/reveal/example/reveal.js]: of course 👌🏻

**Pull request merged**